kernel/src/log_segment.rs

```diff
@@ -37,10 +37,11 @@ mod tests;
 /// and in `TableChanges` when built with [`LogSegment::for_table_changes`].
 ///
 /// [`Snapshot`]: crate::snapshot::Snapshot
-#[derive(Debug)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
 pub(crate) struct LogSegment {
     pub end_version: Version,
+    pub checkpoint_version: Option<Version>,
```

Comment:
this is causing semver check to fail - i've removed flag since this is `pub` only if dev-vis is on

Reply:
also note #810